File: backend/src/main/java/cz/forgottenempire/servermanager/steamauth/AuthVerificationResult.java
package cz.forgottenempire.servermanager.steamauth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Result of Steam authentication verification process.
 * Contains information about the authentication status, type, and any relevant messages.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthVerificationResult {
    
    private AuthStatus status;
    private String message;
    private AuthType authType;
    
    /**
     * Authentication status
     */
    public enum AuthStatus {
        SUCCESS,
        REQUIRES_2FA,
        INVALID_CREDENTIALS,
        ERROR
    }
    
    /**
     * Type of authentication
     */
    public enum AuthType {
        NONE,
        EMAIL,
        MOBILE,
        UNKNOWN
    }
}

File: backend/src/main/kotlin/cz/forgottenempire/servermanager/steamauth/AuthVerificationResult.kt
package cz.forgottenempire.servermanager.steamauth

data class AuthVerificationResult(
    val status: AuthStatus? = null,
    val message: String? = null,
    val authType: AuthType? = null
) {
    enum class AuthStatus {
        SUCCESS, REQUIRES_2FA, INVALID_CREDENTIALS, ERROR
    }

    enum class AuthType {
        NONE, EMAIL, MOBILE, UNKNOWN
    }

    class Builder {
        private var status: AuthStatus? = null
        private var message: String? = null
        private var authType: AuthType? = null

        fun status(status: AuthStatus) = apply { this.status = status }
        fun message(message: String?) = apply { this.message = message }
        fun authType(authType: AuthType) = apply { this.authType = authType }
        fun build() = AuthVerificationResult(status, message, authType)
    }

    companion object {
        @JvmStatic
        fun builder() = Builder()
    }
}
